pkg/tsdb/loki/loki.go

```diff
@@ -93,16 +150,16 @@ func (e *LokiExecutor) DataQuery(ctx context.Context, dsInfo *models.DataSource,
 
 		value, err := client.QueryRange(query.Expr, limit, query.Start, query.End, logproto.BACKWARD, query.Step, interval, false)
 		if err != nil {
-			return plugins.DataResponse{}, err
+			return &backend.QueryDataResponse{}, err
 		}
 
-		queryResult, err := parseResponse(value, query)
+		frames, err := parseResponse(value, query)
 		if err != nil {
-			return plugins.DataResponse{}, err
+			return &backend.QueryDataResponse{}, err
```

Comment:
I saw we have return nil and return newly built struct here, since you create the result at the beginning, might just return the result all the time, could be easier?

Reply:
yes, I agree, going to change it